x-pack/plugin/esql/compute/gen/src/main/java/org/elasticsearch/compute/gen/AggregatorImplementer.java

```diff
@@ -427,14 +428,14 @@ private MethodSpec addRawBlock(boolean masked) {
             if (masked) {
                 builder.beginControlFlow("if (mask.getBoolean(p) == false)").addStatement("continue").endControlFlow();
             }
-            for (AggregationParameter p : aggParams) {
-                builder.addStatement("int $LValueCount = $L.getValueCount(p)", p.name(), p.blockName());
-                builder.beginControlFlow("if ($LValueCount == 0)", p.name());
+            for (Argument a : aggParams) {
+                builder.addStatement("int $LValueCount = $L.getValueCount(p)", a.name(), a.blockName());
+                builder.beginControlFlow("if ($LValueCount == 0)", a.name());
                 builder.addStatement("continue");
                 builder.endControlFlow();
             }
 
-            if (aggParams.getFirst().isArray()) {
+            if (aggParams.getFirst() instanceof BlockArgument) {
```

Comment:
I'm not super happy we have this branch at all, but now is probably not the time to nuke it.